packages/edit-site/src/components/add-new-template/new-template.js

```diff
@@ -229,17 +230,21 @@ function useMissingTemplates(
 		entitiesConfig.tag,
 		onClickMenuItem
 	);
+	const authorMenuItem = useAuthorTemplate( onClickMenuItem );
```

Comment:
Why not use the same hook like the others? I would expect this PR to look more like this: https://github.com/WordPress/gutenberg/pull/42138 for pages. 

The one thing I'm not sure right now(haven't searched in depth) is if/how to get the user type info(labels, etc..). If there isn't a clear way and all the existing strings/info related to users are hardcoded in core (ex in `users.php`, etc..), we should construct such an object with labels used in core.

I would expect to see `useExtraTemplates` here that also handles existing templates etc.. For example in this PR all authors are always suggested without taking into account the already existent specific templates.